### DISCUSSION TITLE
What type of offset is provided by nvidia_get_gem_map_offset?

### DISCUSSION THREAD

**Discussion by YusufKhan-gamedev** [NVIDIA/open-gpu-kernel-modules#249] [Q&A] [answered]
When going through the uapi headers I saw the drm_nvidia_gem_map_offset_params definition:
```
struct drm_nvidia_gem_map_offset_params {
    uint32_t handle;             /* IN Handle to gem object */
    uint32_t __pad;

    uint64_t offset;             /* OUT Fake offset */
};
```
What does "fake" mean in this context?

**Comment by aaronp24** [comment, marked as answer]
For a number of reasons, it's not feasible to map memory pages from a drm 'gem' object into a process's virtual memory space while inside of an ioctl handler. So instead, applications have to use the `mmap` syscall to actually map pages. Applications choose which gem object to map by using an ioctl, and then pass the returned "fake offset" value as the `off` parameter to the `mmap` syscall. The drm driver then uses this offset value to look up the gem object to map.

The core drm code in the kernel does something similar. See [`drm_gem_create_mmap_offset_size`](https://github.com/torvalds/linux/blob/3b5e1590a26713a8c76896f0f1b99f52ec24e72f/drivers/gpu/drm/drm_gem.c#L458) in the kernel.

**Reply by YusufKhan-gamedev** [reply]
analogous to nouveau_bo_priv's map_handle member
